fMRISurface unified with NHP processing

mmilch01 opened the pull request:
FMRISurface adapted for NHP processing.

mmilch01 committed:
+compared GenericfMRISUrfaceProcessingPipeline.sh, RibbonVolumeToSurfaceMapping.sh, SurfaceSmoothing.sh, SubcorticalProcessing.sh

mmilch01 committed:
+example NHP batch

glasserm commented on Examples/Scripts/GenericfMRISurfaceProcessingPipelineBatchNHP.sh:
```diff
@@ -0,0 +1,166 @@
+StudyFolder="${HOME}/projects/Pipelines_ExampleData" #Location of Session folders (named by SessionID)
+Sesslist="100307 100610" #Space delimited list of Session IDs
+EnvironmentScript="${HOME}/projects/Pipelines/Examples/Scripts/SetUpHCPPipeline.sh" #Pipeline environment script
+SPECIES="Human"
```
For NHP batch I would default to something other than human.  Perhaps Macaque.

glasserm commented on fMRISurface/scripts/SubcorticalProcessing.sh:
```diff
@@ -126,11 +136,18 @@ rm -f ${ResultsFolder}/${NameOffMRI}_temp_template.dlabel.nii
 
 #write output volume, delete temporary
 #NOTE: $VolumefMRI contains a path in it, it is not a file in the current directory
+#TODO check NHP: NHP version doesn't have '_dilate' suffix in the command below - supposedly because no dilation is done. Resolve before merging.
```
I don't understand this comment.

glasserm replied:
Misha will elaborate and make changes if needed.

coalsont commented on fMRISurface/scripts/SubcorticalProcessing.sh:
```diff
@@ -99,7 +104,12 @@ fi
+    ${CARET7DIR}/wb_command -cifti-dilate ${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii COLUMN 0 10 ${ResultsFolder}/${NameOffMRI}_temp_subject_dilate.dtseries.nii -nearest -merged-volume
+else
+    ${CARET7DIR}/wb_command -cifti-dilate ${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii COLUMN 0 30 ${ResultsFolder}/${NameOffMRI}_temp_subject_dilate.dtseries.nii
+fi
```
Here is my suggested improvement to do within-structure dilation first, so that any data that doesn't need `-merged-volume` to get rid of the zeros doesn't bleed data across structures.  I am leaving in `-nearest` under the assumption that it may have been done for speed (and because weighted wouldn't be substantially closer to salvaging a masked-out structure anyway).

I imagine we don't want this 2-step process in human, because if the brain mask excludes an entire subcortical structure, the subject's data quality needs to be checked.

There is a second dilation section below with similar logic, which should presumably be updated the same way.
```suggestion
if (( NonHumanSpecies )); then
    "$CARET7DIR"/wb_command -cifti-dilate "${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii" COLUMN 0 10 "${ResultsFolder}/${NameOffMRI}_temp_subject_dilate_init.dtseries.nii"
    "$CARET7DIR"/wb_command -cifti-dilate "${ResultsFolder}/${NameOffMRI}_temp_subject_dilate_init.dtseries.nii" COLUMN 0 10 "${ResultsFolder}/${NameOffMRI}_temp_subject_dilate.dtseries.nii" -nearest -merged-volume
    rm -f "${ResultsFolder}/${NameOffMRI}_temp_subject_dilate_init.dtseries.nii"
else
    "$CARET7DIR"/wb_command -cifti-dilate "${ResultsFolder}/${NameOffMRI}_temp_subject.dtseries.nii" COLUMN 0 30 "${ResultsFolder}/${NameOffMRI}_temp_subject_dilate.dtseries.nii"
fi
```

mmilch01 committed:
Update fMRISurface/scripts/SubcorticalProcessing.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Update Examples/Scripts/GenericfMRISurfaceProcessingPipelineBatchNHP.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

mmilch01 committed:
Update fMRISurface/scripts/SubcorticalProcessing.sh

Co-authored-by: Tim Coalson <coalsont@users.noreply.github.com>

Commit:
+Address comments:
https://github.com/Washington-University/HCPpipelines/pull/370/files#r2582499018
https://github.com/Washington-University/HCPpipelines/pull/370/files#r2582503918
https://github.com/Washington-University/HCPpipelines/pull/370/files#r2582955420

coalsont commented on fMRISurface/scripts/SubcorticalProcessing.sh:
```diff
@@ -126,7 +139,12 @@ rm -f ${ResultsFolder}/${NameOffMRI}_temp_template.dlabel.nii
+
+if (( NonHumanSpecies )); then 
+    ${CARET7DIR}/wb_command -cifti-dilate ${ResultsFolder}/${NameOffMRI}_temp_atlas.dtseries.nii COLUMN 0 10 ${ResultsFolder}/${NameOffMRI}_temp_atlas_dilate.dtseries.nii -nearest -merged-volume
+else
```
Maybe this should match the other dilation section above.

mmilch01 replied:
Addressed in 78613ea.

Commit:
+address reviewer comments
https://github.com/Washington-University/HCPpipelines/pull/370#discussion_r2600818146
https://github.com/Washington-University/HCPpipelines/pull/370#discussion_r2600824256

Commit:
+Added command line specified arguments

glasserm reviewed (approved):
Testing passed.
